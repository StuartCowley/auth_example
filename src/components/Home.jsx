import "../styles/home.css"
import Header from "./Header"

const Home = () => {
    return (
        <div className="home">
            <Header />
            <p>This is the homepage</p>
        </div>
    )
}

export default Home