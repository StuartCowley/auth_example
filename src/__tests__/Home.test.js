import { BrowserRouter as Router } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from '../components/Home';

describe("Home component", () => {
    it("matches snapshot", () => {
        const generated = renderer
            .create(
                <Router>
                    <Home />
                </Router>
            ).toJSON();

        expect(generated).toMatchSnapshot();
    });

    it("displays welcome message", () => {
        render(
            <Router>
                <Home />
            </Router>
        );

        expect(screen.getByText("This is the homepage")).toBeInTheDocument;
    })
});