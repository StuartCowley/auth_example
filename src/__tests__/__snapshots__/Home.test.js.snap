// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div
  className="home"
>
  <nav
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <div
      className="header__nav-link"
      onClick={[Function]}
    >
      Account
    </div>
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </nav>
  <p>
    This is the homepage
  </p>
</div>
`;
